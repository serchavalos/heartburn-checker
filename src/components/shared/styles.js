export const minWidth = "500px";
export const fontColorPrimary = "#3f6072";
export const fontColorSecondary = "#7b98a9";
export const green = "#6accba";
export const gray = "#e8ecf0";
