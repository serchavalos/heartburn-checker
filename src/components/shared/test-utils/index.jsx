import { MemoryRouter } from "react-router-dom";

export const TestContainer = MemoryRouter;
