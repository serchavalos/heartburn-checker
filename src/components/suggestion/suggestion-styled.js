import styled from "styled-components";
import { Link as BaseLink } from "react-router-dom";

import { green } from "../shared/styles";

export const Container = styled.div`
  margin-top: 7rem;
`;

export const Link = styled(BaseLink)`
  color: ${green};
  display: block;
  width: 100%;
  text-align: center;

  &:visited {
    color: ${green};
  }
`;
