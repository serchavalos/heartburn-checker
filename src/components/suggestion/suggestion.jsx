import React from "react";
import PropTypes from "prop-types";

import { Title } from "../shared/title/title";
import { CtaButton } from "../shared/button/button";
import { Container, Link } from "./suggestion-styled";

export const Suggestion = ({ text, showBookingButton }) => {
  return (
    <React.Fragment>
      <Container>
        <Title>Thank for answering the questions!</Title>
        <p>{text}</p>
        {showBookingButton && (
          <CtaButton href="https://www.kry.se/" target="_blank">
            Booking a meeting
          </CtaButton>
        )}
      </Container>
      <Link to="/">Back to the start screen</Link>
    </React.Fragment>
  );
};

Suggestion.propTypes = {
  text: PropTypes.string,
  showBookingButton: PropTypes.bool
};
