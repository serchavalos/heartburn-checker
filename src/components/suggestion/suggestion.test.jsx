import React from "react";
import { render } from "react-testing-library";

import { TestContainer } from "../shared/test-utils";
import { Suggestion } from "./suggestion.jsx";

describe("QuestionForm", () => {
  it("should render propperly", () => {
    const { container, getByText } = render(
      <TestContainer>
        <Suggestion
          text="You seem to be dying but you'll be fine"
          showBookingButton={true}
        />
      </TestContainer>
    );
    expect(container).toMatchSnapshot();
    expect(getByText("You seem to be dying but you'll be fine")).toBeDefined();
    expect(container.querySelectorAll("a").length).toBe(2);
  });

  it("should *not* render the CTA button", () => {
    const { container, getByText } = render(
      <TestContainer>
        <Suggestion text="Just veggies" showBookingButton={false} />
      </TestContainer>
    );
    expect(getByText("Just veggies")).toBeDefined();
    expect(container.querySelectorAll("a").length).toBe(1);
  });
});
