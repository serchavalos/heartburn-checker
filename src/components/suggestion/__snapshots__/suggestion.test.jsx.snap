// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionForm should render propperly 1`] = `
<div>
  <div
    class="sc-bxivhb luIqf"
  >
    <h1
      class="sc-bdVaJa eAcTCg"
    >
      Thank for answering the questions!
    </h1>
    <p>
      You seem to be dying but you'll be fine
    </p>
    <a
      class="sc-htpNat iTsvih"
      href="https://www.kry.se/"
      target="_blank"
    >
      Booking a meeting
    </a>
  </div>
  <a
    class="sc-ifAKCX cuLtBW"
    href="/"
  >
    Back to the start screen
  </a>
</div>
`;
